testnet/stacks-node/src/tests/neon_integrations.rs

```diff
@@ -1127,3 +1155,531 @@ fn pox_integration_test() {
+                };
+                serde_json::to_vec(&json!(content)).unwrap()
+            };
+
+            let path = format!("{}/v2/transactions", &http_origin);
+            let res = client
+                .post(&path)
+                .header("Content-Type", "application/json")
+                .body(body)
+                .send()
+                .unwrap();
+            eprintln!("{:#?}", res);
+            if !res.status().is_success() {
+                eprintln!("{}", res.text().unwrap());
+                panic!("");
+            }
+        }
+
+        bootstrap_node_tx
+            .send(Signal::ReplicatingAttachmentsStartTest2)
+            .expect("Unable to send signal");
+
+        let target_height = match follower_node_rx.recv() {
+            Ok(Signal::ReplicatingAttachmentsCheckTest2(target_height)) => target_height,
+            _ => panic!("Bootstrap node could nod boot. Aborting test."),
+        };
+
+        let mut sort_height = channel.get_sortitions_processed();
+        while sort_height < target_height {
+            wait_for_runloop(&blocks_processed);
+            sort_height = channel.get_sortitions_processed();
+        }
+
+        // Poll GET v2/attachments/<attachment-hash>
+        for i in 1..10 {
+            let mut attachments_did_sync = false;
+            let mut timeout = 30;
+            while attachments_did_sync != true {
+                let zonefile_hex = hex_bytes(&format!("facade0{}", i)).unwrap();
+                let hashed_zonefile = Hash160::from_data(&zonefile_hex);
+                let path = format!(
+                    "{}/v2/attachments/{}",
+                    &http_origin,
+                    hashed_zonefile.to_hex()
+                );
+                let res = client
+                    .get(&path)
+                    .header("Content-Type", "application/json")
+                    .send()
+                    .unwrap();
+                eprintln!("{:#?}", res);
+                if res.status().is_success() {
+                    let attachment_response: GetAttachmentResponse = res.json().unwrap();
+                    assert_eq!(attachment_response.attachment.content, zonefile_hex);
+                    attachments_did_sync = true;
+                } else {
+                    timeout -= 1;
+                    if timeout == 0 {
+                        panic!("Failed syncing 9 attachments between 2 neon runloops within 30s - Something is wrong");
+                    }
+                    eprintln!("Attachment {} not sync'd yet", bytes_to_hex(&zonefile_hex));
+                    thread::sleep(Duration::from_millis(1000));
+                }
+            }
+        }
+
+        // Ensure that we the attached sidecar was able to receive a total of 10 attachments
+        assert_eq!(test_observer::get_attachments().len(), 10);
+
+        channel.stop_chains_coordinator();
+    });
+
+    bootstrap_node_thread.join().unwrap();
+    follower_node_thread.join().unwrap();
```

Comment:
What you have here is fine, but can you open an issue to track testing this with up to 1000 names with attachments?

Reply:
Issue created https://github.com/blockstack/stacks-blockchain/issues/2142